packages/iov-lisk/src/liskclient.ts

```diff
@@ -0,0 +1,200 @@
+    const url = this.baseUrl + "/api/node/status";
+    const result = await axios.get(url);
+    const responseBody = result.data;
+    return responseBody.data.height;
+  }
+
+  public async postTx(bytes: PostableBytes): Promise<BcpTransactionResponse> {
+    const transactionId = JSON.parse(Encoding.fromUtf8(bytes)).id as string;
+    if (!transactionId.match(/^[0-9]+$/)) {
+      throw new Error("Invalid transaction ID");
+    }
+
+    await axios.post(this.baseUrl + "/api/transactions", bytes, {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    // Sleep some seconds to ensure transaction will be found.
+    // Sleep duration determined by trial and error. 15 seconds was not enough.
+    await new Promise(resolve => setTimeout(resolve, 20 * 1000));
+
+    const result = await axios.get(this.baseUrl + `/api/transactions?id=${transactionId}`);
+    const responseBody = result.data;
+
+    let height: number | undefined;
+    let transactionResultBytes: Uint8Array | undefined;
+    if (responseBody.meta.count === 1) {
+      const transactionResult = responseBody.data[0];
+      height = transactionResult.height;
+      transactionResultBytes = Encoding.toUtf8(JSON.stringify(transactionResult));
+    }
+
+    return {
+      metadata: {
+        status: true, // commit failures should throw
+        height: height,
+      },
+      data: {
+        message: "",
+        txid: Encoding.toAscii(transactionId) as TxId,
+        result: transactionResultBytes || new Uint8Array([]),
+      },
+    };
+  }
+
+  public getTicker(_: TokenTicker): Promise<BcpQueryEnvelope<BcpTicker>> {
+    throw new Error("Not implemented");
+  }
+
+  public getAllTickers(): Promise<BcpQueryEnvelope<BcpTicker>> {
+    throw new Error("Not implemented");
+  }
+
+  public async getAccount(query: BcpAccountQuery): Promise<BcpQueryEnvelope<BcpAccount>> {
+    if (isAddressQuery(query)) {
+      const address = query.address;
+      const url = this.baseUrl + `/api/accounts?address=${Encoding.fromAscii(address)}`;
+      const result = await axios.get(url);
+      const responseBody = result.data;
+
+      const account: BcpAccount = {
+        address: address,
+        name: undefined,
+        balance: [
+          {
+            sigFigs: 8,
+            tokenName: undefined,
+            ...Parse.liskAmount(responseBody.data[0].balance),
+          },
+        ],
+      };
+
+      const wrapped: BcpQueryEnvelope<BcpAccount> = {
+        metadata: {
+          offset: 0,
+          limit: 0,
+        },
+        data: [account],
+      };
+
+      return wrapped;
+    } else {
+      throw new Error("Query type not supported");
+    }
+  }
+
+  public getNonce(query: BcpAccountQuery): Promise<BcpQueryEnvelope<BcpNonce>> {
+    if (isAddressQuery(query)) {
+      const address = query.address;
+
+      const nonce: BcpNonce = {
+        address: address,
+        // fake pubkey, we cannot always know this
```

Comment:
do we ever get to see the pubkey?
i mean, after they sign one transaction it should be visible somehow?

empty is fine if they never signed

Reply:
We can query the API and will get a pubkey if and only of the address did a transaction before. However, this adds another network query, will not work for the first outgoing transaction of an address and the result is thrown away anyway.